apps/patient/src/components/ReadyText.tsx

```diff
@@ -0,0 +1,106 @@
+import dayjs, { Dayjs } from 'dayjs';
+import { Text } from '@chakra-ui/react';
+import { OrderFulfillment } from 'packages/sdk/dist/types';
+
+export type Maybe<T> = T | null;
+interface ReadyTextProps {
+  readyBy?: string;
+  readyByDay?: string;
+  isDeliveryPharmacy?: boolean;
+  fulfillment?: Maybe<OrderFulfillment> | undefined;
+}
+const ReadyText = ({ readyBy, readyByDay, isDeliveryPharmacy, fulfillment }: ReadyTextProps) => {
+  const showPatientDesiredreadyBy =
+    readyBy &&
+    readyByDay &&
+    !isDeliveryPharmacy &&
+    (!fulfillment || // No fulfillment means user came from pharmacy selection
+      fulfillment?.state === 'SENT');
+  const showEstimatedReadyAt =
+    fulfillment?.pharmacyEstimatedReadyAt &&
+    !isDeliveryPharmacy &&
+    (!fulfillment || // No fulfillment means user came from pharmacy selection
+      fulfillment?.state === 'RECEIVED');
+
+  return (
+    <>
+      {showPatientDesiredreadyBy && (
+        <PatientDesiredReadyBy readyBy={readyBy} readyByDay={readyByDay} />
+      )}
+      {showEstimatedReadyAt && (
+        <PharmacyEstimatedReadyAt pharmacyEstimatedReadyAt={fulfillment.pharmacyEstimatedReadyAt} />
+      )}
+    </>
+  );
+};
```

Comment:
Its hard to parse this. I _think_ this is:
- If deliveryPharmacy = do nothing
- If SENT = show desired
- If RECEIVED = show estimated ready

Is that right?
If yes, I think we can simplify this to:

```suggestion
  if (isDeliveryPharmacy) return null;
  
  if ((!fulfillment || fulfillment?.state === 'SENT') && readyBy && readyByDay) {
    return <PatientDesiredReadyBy readyBy={readyBy} readyByDay={readyByDay} />
  }
  
  if (fulfillment?.state === 'RECEIVED' && fulfillment?.pharmacyEstimatedReadyAt) {
    return <PharmacyEstimatedReadyAt pharmacyEstimatedReadyAt={fulfillment.pharmacyEstimatedReadyAt} />
  }
```

This makes it clearer that only one of these is being returned and when.

Reply:
oh nice, way better